app/src/main/java/com/codesoom/assignment/application/ProductService.java

```diff
@@ -4,61 +4,92 @@
 import com.codesoom.assignment.domain.Product;
 import com.codesoom.assignment.domain.ProductRepository;
 import com.codesoom.assignment.dto.ProductData;
+import com.github.dozermapper.core.Mapper;
 import org.springframework.stereotype.Service;
 
 import javax.transaction.Transactional;
 import java.util.List;
 
+/**
+ * 상품 도메인의 서비스 레이어.
+ */
 @Service
 @Transactional
 public class ProductService {
+    private final Mapper mapper;
     private final ProductRepository productRepository;
 
-    public ProductService(ProductRepository productRepository) {
+    /**
+     * ProductService 생성자.
+     *
+     * @param productRepository 상품 도메인의 퍼시스턴스 레이어.
+     * @param mapper            객체 매퍼.
+     */
+    public ProductService(ProductRepository productRepository,
+                          Mapper mapper) {
         this.productRepository = productRepository;
+        this.mapper = mapper;
     }
 
+    /**
+     * 상품 목록을 조회한다.
+     *
+     * @return 상품 목록.
+     */
```

Comment:
JavaDoc에서는 리턴 값이 중요합니다. `String.class`나 `File.class` 같은 파일들의 JavaDoc을 꼭 읽어보세요. 다음과 같은 두 대안을 생각할 수 있겠죠.
```suggestion
    /**
     * 상품 목록을 조회해 리턴한다.
     *
     * @return 상품 목록
     */
```

```suggestion
    /**
     * 상품 목록을 리턴한다.
     *
     * @return 상품 목록
     */
```

Reply:
`@return` 태그가 있어서 무엇을 리턴하는지는 다시 설명하지 않아도 된다고 생각했었습니다. 표준 라이브러리를 보면 `@return` 태그는 간단하게 작성하고 설명문에 좀 더 풀어쓰는 경향이 있는 것 같습니다. 하지만 무엇을 리턴하는지를 왜 중복해서 작성해야 하는지는 모르겠습니다. 간단하게 살펴봤지만 오픈 소스는 커뮤니티마다 문서 작성 스타일이 다른 것 같아서 무엇이 정답이라고 할 수도 없는 것 같습니다. 아직은 종립님의 지식을 얻어가면서 조금씩 깨달아가겠습니다! 리뷰 감사드립니다.

Reply:
왜 리턴값에 대한 설명을 두 번 해야 할까요? 사실 그 이유는 되게 단순합니다. 두 문장의 역할이 다르기 때문입니다. 첫번째 문장은 메소드의 기능과 역할을 설명하는 것이고, 두 번째 문장은 리턴값을 설명하는 것입니다. 그래서 이 메소드처럼 단순한 메소드는 두 문장이 중복이 되기도 해요. 저는 이렇게 중복이 되는 경우는 그냥 description만 남겨두기도 합니다. 지금 강조하는 것은 JavaDoc을 작성하는 훈련을 위한 것이고요. 

한편, 조금 더 다른 관점에서 이유를 생각할 수도 있는데요, JavaDoc으로 웹 페이지를 빌드하면 화면 최상단에는 description만 나오고, `@return` 태그는 보이지 않기 때문이기도 합니다. 

String.class 의 JavaDoc을 함께 읽어봅시다.

https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html

![image](https://user-images.githubusercontent.com/1855714/120482992-a69df500-c3ec-11eb-914f-97a9ec505a5b.png)

메소드 상세 내역을 보면 이렇게 보입니다.

![image](https://user-images.githubusercontent.com/1855714/120483054-b6b5d480-c3ec-11eb-9d60-1d3f0aad5683.png)

`@return`의 설명이 복잡한 메소드의 예로는 `compareTo`가 있습니다. 잘 읽어보세요. 느낌이 꽤 다르죠?

![image](https://user-images.githubusercontent.com/1855714/120483238-dfd66500-c3ec-11eb-867d-693db73570b1.png)

아직은 저를 믿고 깨달아가겠다는 말씀이 감사합니다.
제가 특별히 강조하는 것들은 자바 스탠다드 라이브러리에 있는 것들이기도 합니다. 가급적이면 공식 스타일을 따라 학습하길 바라는 마음입니다.

Reply:
다시 보니 [구글 자바 스타일 가이드#7.2](https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment)에도 `/** @return the customer ID */` 대신 `/** Returns the customer ID. */` 로 작성하라는 것이 말씀해주신 것과 같은 맥락이었네요! (여기서 말하는 The summary fragment가 자바독 상단에 있는 Summary였군요...)

앞으로 공식 JavaDoc을 단순히 훑는 것이 아니라 스타일까지 살펴보는 습관을 갖겠습니다.